### DISCUSSION TITLE
백엔드 인프라 마이그레이션

### DISCUSSION THREAD

**Discussion by nayonsoso** [woowacourse-teams/2024-review-me#960] [문서화]
### 선호하는 클라우드 컴퓨팅 서비스 순위

**1위 : azure**

- 저렴함
  - 학생 크레딧 14만원으로 넉넉한 서버 구성 가능
  - aws 도 프리티어를 사용하면 비용을 절약할 수 있지만, 서비스 스펙에 한계가 있음
  - aws 에서 월 3.6달러였던 EIP(고정 IP)도 azure 에서는 무료로 사용 가능 (azure 에서는 사용하지 않는 IP만 금액이 청구됨)
- aws 다음으로 레퍼런스가 많음
- (개인적인 호기심) aws 말고 다른 클라우드 컴퓨팅 서비스는 어떻게 구성되는지 궁금함..

**2위 : aws**
- 우리 모두가 알고있음
  - 매월 청구되는 금액 (4000~12000)이 더 클까, 우리가 새로운 클라우드 컴퓨팅을 배워서 적용하는 것이 클까?

나머지는 다 메리트가 없다고 느껴짐

### 인프라 구조도

AS-IS

![image](https://github.com/user-attachments/assets/041c2954-85da-4896-8379-68f03b308b73)

TO-BE

![image](https://github.com/user-attachments/assets/3114347f-8444-4478-98e3-c7b78e6e2988)

- 인프라에서의 '가용성'을 고려하기 전에, 사용자가 쓸만한 서비스가 되어야 한다고 생각한다.
- 예를들면 우테코 7기에서 사용할만한 서비스가 되도록 서비스 측면에서 더 업그레이드를 해야한다.
- 그리고 사용자가 증가함에 따라서 문제가 생긴다면, 그때 인프라 확장을 생각해보자.
- 이렇게 인프라 확장을 경계짓는 이유 : _**인프라는 확장 생각하면 끝도 없음**_

**Comment by donghoony** [comment]
Oracle: Arm 인스턴스 2대 (2코어 12GB Ram) 무료 사용 가능 ㅎㅎ;;

**Comment by nayonsoso** [comment]
### 🧑🏻‍⚖️ 회의 결과 

- 사용할 서비스 : 오라클
- 인프라 구조도 : 
![image](https://github.com/user-attachments/assets/7d22b6b3-1991-4acb-b4ca-468907b17b00)

- redis 에 대해서 : 인메모리 데이터베이스인 카페인도 레디스와 동일하게 TTL 을 지원하긴 한다.
그런데 굳이 redis에서 카페인으로 옮길 이유도 없는 상황 ...
따라서 레디스는 유지하기로 했다.

**Comment by hyeonji1220** [comment]
[산초]
Oracle 계정 생성 & 공유
인터넷 게이트웨이, nat, 라우팅 테이블, 서브넷, 보안그룹 설정
ec2 생성 및 ssh 접속에 필요한 pem 공유

[테드]
dev ec2 / prod db ec2에 mysql 설치
기존 데이터 이전

[아루]
ci/cd 변경 내용 수정
모니터링 ec2에 필요한 서비스 설치 & 모니터링 구성 수정

[커비]
prod ec2 / dev ec2 에 필요한 서비스 설치
cd 돌려서 이전 사항들이 잘 작동하는지 확인

[이후 아루가 해줘야 할 일]
프론트에까지 우리 ip 변경된 것을 알려준 다음,
cloud fare 에서 도메인과 연결된 ip 변경하기
